plugins/syntax/g_string.mlg

```diff
@@ -21,5 +21,6 @@ open Stdarg
 VERNAC COMMAND EXTEND StringNotation CLASSIFIED AS SIDEFF
   | #[ locality = Attributes.locality; ] [ "String" "Notation" reference(ty) reference(f) reference(g) ":"
       ident(sc) ] ->
-    { vernac_string_notation (Locality.make_module_locality locality) ty f g (Id.to_string sc) }
+    { let (sigma, env) = Pfedit.get_current_context () in
```

Comment:
@maximedenes this is a good example of a dubious use of `get_current_context`, this function needs the global environment, and here it only happens to work because get_current_context will never be called inside a proof.

Reply:
Yes, that's what I thought too, but I tried to limit this PR to making the threading explicit, without changing too much the existing code.

We should however keep track of the problematic occurrences, maybe.

Reply:
My point of view here is that is safer [and easier to track] to make such functions not to take an env sigma, as they are not supposed to take it, and have them just use the global environment.

Having `env -> evar_map` is extremely confusing as it gives the user the wrong impression.